src/middlewared/middlewared/plugins/disk_/configure_swap.py

```diff
@@ -0,0 +1,206 @@
+                # to avoid errors in the console (#27516)
+                await run('savecore', '-z', '-m', '5', '/data/crash/', f'/dev/{swap_part["name"]}', check=False)
+
+            if swap_part['disk'] in disks:
+                swap_partitions_by_size[swap_part['size']].append(swap_part['name'])
+
+        dumpdev = False
+        unused_partitions = []
+        for size, partitions in swap_partitions_by_size.items():
+            # If we have only one partition add it to unused_partitions list
+            if len(partitions) == 1:
+                unused_partitions += partitions
+                continue
+
+            for i in range(int(len(partitions) / 2)):
+                if (
+                    len(create_swap_devices) + len(existing_swap_devices['mirrors']) +
+                        len(existing_swap_devices['partitions'])
+                ) > MIRROR_MAX:
+                    break
+                part_ab = partitions[0:2]
+                partitions = partitions[2:]
+
+                # We could have a single disk being used as swap, without mirror.
+                try:
+                    for p in part_ab:
+                        remove = False
+                        if IS_LINUX:
+                            part = all_partitions[p]['path']
+                            if part in existing_swap_devices['partitions']:
+                                remove = True
+                        else:
+                            part = p
+                            if part in existing_swap_devices['partitions']:
+                                remove = True
+                            elif f'{part}.eli' in existing_swap_devices['partitions']:
+                                part = f'{part}.eli'
+                                remove = True
+
+                        if remove:
+                            await self.middleware.call('disk.swaps_remove_disks', [all_partitions[p]['disk']])
+                            existing_swap_devices['partitions'].remove(part)
+                except Exception:
+                    self.logger.warn('Failed to remove disk from swap', exc_info=True)
+                    # If something failed here there is no point in trying to create the mirror
+                    continue
+
+                part_a, part_b = part_ab
+
+                if not IS_LINUX and not dumpdev:
+                    dumpdev = await self.middleware.call('disk.dumpdev_configure', part_a)
+                name = await self.middleware.call('disk.new_swap_name')
+                if not name:
+                    # Which means maximum has been reached and we can stop
+                    break
+                part_a_path, part_b_path = all_partitions[part_a]['path'], all_partitions[part_b]['path']
+                if IS_LINUX:
+                    cp = await Popen(
+                        f'echo "y" | mdadm --create {os.path.join("/dev/md", name)} '
```

Comment:
In FreeBSD we dont write blocks in the disk so its not seen after reboot and make things simpler. Is there a way to do that in linux?

Reply:
Yes it should be with `--build`, I'll update.